src/boxed.rs

```diff
@@ -457,6 +466,18 @@ impl<T: 'static, MM: BoxedMemoryManager<T>> FromGlibPtrFull<*mut T> for Boxed<T,
     }
 }
 
+impl<T: 'static, MM: BoxedMemoryManager<T>> FromGlibPtrFull<*const T> for Boxed<T, MM> {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *const T) -> Self {
+        assert!(!ptr.is_null());
+        let ptr = MM::copy(ptr);
+        Boxed {
+            inner: AnyBox::Native(Box::from_raw(ptr)),
```

Comment:
This is only allowed if the pointer was originally a `Box` and allocated from Rust. That's not the case here :)

Also simply copying the memory behind the pointer is not enough here. You'd leak memory.

What you want here is to call the `*mut _` version of `from_glib_full()` by simply casting the pointer.

Reply:
Thanks! Will try to get that in place.

Reply:
Simplest proposal:
`unsafe fn from_glib_full(ptr: *const T) -> Self {
        assert!(!ptr.is_null());
        let ptr = ptr as *mut _;
        Boxed {
            inner: AnyBox::ForeignOwned(ptr::NonNull::new_unchecked(ptr)),
            _dummy: PhantomData,
        }
    }`
This is essentially the same as the FromGlibPtrFull on *mut but with cast to *mut. Would this suffice?

Reply:
You can do it even easier, but basically yes. You could do `from_glib_full(ptr as *mut T)`

Reply:
Could we even generalize over *mut/*const pointers by using a suitable pointer trait in the signature? I quickly tried Ptr but run into trouble.

Reply:
I don't think so